kubel.el

```diff
@@ -934,16 +933,14 @@ P can be a single number or a localhost:container port pair."
   (require 'vterm)
   (interactive)
   (kubel-setup-tramp)
-  (let* ((dir-prefix (or
-                      (when (tramp-tramp-file-p default-directory)
-                        (with-parsed-tramp-file-name default-directory nil
-                          (format "%s%s:%s@%s|" (or hop "") method user host))) ""))
+  (let* ((dir-prefix (kubel--dir-prefix))
          (con-pod (kubel--get-container-under-cursor))
          (container (car con-pod))
          (pod (cdr con-pod))
          (default-directory (format "/%skubectl:%s@%s:/" dir-prefix container pod))
-         (vterm-buffer-name (format "*kubel - vterm - %s@%s*" container pod)))
-  (vterm)))
+         (vterm-buffer-name (format "*kubel - vterm - %s@%s*" container pod))
+         (vterm-shell "/bin/sh"))
```

Comment:
This was really the problem. Vterm probably tries to use your local shell but docker image might not have it. This is the safest way to ensure the problem doesn't happen. As a future enhancement, could try to be smarter and pick bash if available.

ref https://github.com/akermu/emacs-libvterm/issues/542